File: mpi/com.c
#include <stdio.h>
#include <stdlib.h>
#include <mpi.h>
#include <math.h>
 
/************************************************************
This is a simple hello world program. Each processor prints out 
it's rank and the size of the current MPI run (Total number of
processors).
************************************************************/
int main(argc,argv)
int argc;
char *argv[];
{
    int myid, numprocs,aflag;
 
    MPI_Init(&argc,&argv);
    MPI_Comm_size(MPI_COMM_WORLD,&numprocs);
    MPI_Comm_rank(MPI_COMM_WORLD,&myid);
    MPI_Comm_test_inter(MPI_COMM_WORLD, &aflag);

    /* print out my rank and this run's PE size */
    printf("Hello from task %d of %d %d \n",myid,numprocs,aflag);

    MPI_Finalize();
    
}

File: mpi/com.f90
!*********************************************
!  This is a simple hello world program. Each processor 
!  prints out its rank and total number of processors 
!  in the current MPI run. 
!****************************************************************
      program hello

      include "mpif.h"
      logical aflag

      call MPI_INIT( ierr )
      call MPI_COMM_RANK( MPI_COMM_WORLD, myid, ierr )
      call MPI_COMM_SIZE( MPI_COMM_WORLD, numprocs, ierr )
      call MPI_COMM_TEST_INTER(MPI_COMM_WORLD, aflag, ierr)

      write (*,'("Hello from task ",i4," of ",i4,l10)')myid,numprocs,aflag

      call MPI_FINALIZE(ierr)
      stop
      end



